homeassistant/components/analytics/analytics.py

```diff
@@ -306,3 +337,37 @@ async def send_analytics(self, _: datetime | None = None) -> None:
             LOGGER.error(
                 "Error sending analytics to %s: %r", ANALYTICS_ENDPOINT_URL, err
             )
+
+    @callback
+    def _async_validate_integration(
+        self,
+        integration: Integration,
+        yaml_domains: set[str],
+        entity_registry_platforms: set[str],
+    ) -> bool:
+        """Return a bool to indicate if this integration should be reported."""
+        if integration.disabled:
+            return False
+
+        # Check if the integration is defined in YAML or in the entity registry
+        if (
+            integration.domain not in yaml_domains
+            and integration.domain not in entity_registry_platforms
+        ):
+            # Check if the integration provide a config flow
+            if not integration.config_flow:
+                return False
+
+            entries = self.hass.config_entries.async_entries(integration.domain)
+
+            # Filter out ignored and disabled entries
+            active_entries = [
+                entry
+                for entry in entries
+                if entry.source != SOURCE_IGNORE and entry.disabled_by is None
+            ]
+
+            if len(active_entries) == 0:
+                return False
```

Comment:
instead of creating a whole list and then checking the length, you can use

```python
return any(
    entry
    for entry in entries
    if entry.source != SOURCE_IGNORE and entry.disabled_by is None
)
```